js/src/javascript/beautifier.js

```diff
@@ -664,6 +668,44 @@ function Beautifier(js_source_text, options) {
         output.space_before_token = opt.space_after_anon_function;
       }
 
+
+      // {
+      //   data() {}
+      // };
+      // vs
+      // {
+      //   data () {}
+      // }
+      if (current_token.text === '(') {
+        if (last_type === 'TK_RESERVED' && in_array(flags.last_word, ['get', 'set'])) {
+          output.space_before_token = opt.space_after_function;
+        }
+        if (last_type === 'TK_WORD' &&
+          tokens[token_pos - 2] &&
+          !in_array(tokens[token_pos - 2].type, ['TK_DOT']) &&
+          (previous_flags.mode === 'Expression' || tokens[token_pos - 2].parent && tokens[token_pos - 2].parent.type === 'TK_EQUALS')
+        ) {
+          output.space_before_token = opt.space_after_function;
+        }
+        if (last_type === 'TK_WORD' &&
+          tokens[token_pos - 2] &&
+          !in_array(tokens[token_pos - 2].type, ['TK_DOT', 'TK_RESERVED', 'TK_END_BLOCK', 'TK_END_EXPR', 'TK_SEMICOLON']) &&
+          ((previous_flags.mode !== 'Statement' || previous_flags.mode !== 'BlockStatement') &&
+            (previous_flags.declaration_statement || flag_store[flag_store.length - 1].parent.last_text === 'default') ||
+            previous_flags.parent.mode === 'ObjectLiteral') &&
+          previous_flags.parent &&
+          previous_flags.parent.mode !== 'Statement' &&
```

Comment:
Use constants such as `TOKEN` and `MODE`.

Reply:
The modified